app/models/work_package/exports/macros/attributes.rb

```diff
@@ -145,18 +150,30 @@ def self.resolve_value_work_package(work_package, attribute)
       end
 
       def self.resolve_value(obj, attribute, disabled_rich_text_fields)
-        cf = obj.available_custom_fields.find { |pcf| pcf.name == attribute }
+        custom_field = find_custom_field(obj, attribute)
+        return msg_macro_error_rich_text if custom_field&.formattable?
 
-        return msg_macro_error_rich_text if cf&.formattable?
+        attribute_name = convert_to_attribute_name(custom_field, attribute, obj)
+        return "" unless can_view_attribute?(custom_field, obj, attribute_name)
```

Comment:
Does it need to be a string? Can it be `nil`:
```suggestion
        return unless can_view_attribute?(custom_field, obj, attribute_name)
```

Reply:
No, even more, empty string would be a possible bug in markdown, so I return a space character instead now.
https://github.com/opf/openproject/pull/19449/commits/38e924b0f3780d4637744ea8cd6d89a6644497f0